packages/xrpl/src/models/utils/flags.ts

```diff
@@ -65,55 +58,33 @@ export function setTransactionFlagsToNumber(tx: Transaction): void {
 
   switch (tx.TransactionType) {
     case 'AccountSet':
-      tx.Flags = convertAccountSetFlagsToNumber(tx.Flags)
+      tx.Flags = convertFlagsToNumber(tx.Flags, AccountSetTfFlags)
+      return
+    case 'AMMDeposit':
+      tx.Flags = convertFlagsToNumber(tx.Flags, AMMDepositFlags)
+      return
+    case 'AMMWithdraw':
+      tx.Flags = convertFlagsToNumber(tx.Flags, AMMWithdrawFlags)
       return
     case 'OfferCreate':
-      tx.Flags = convertOfferCreateFlagsToNumber(tx.Flags)
+      tx.Flags = convertFlagsToNumber(tx.Flags, OfferCreateFlags)
       return
     case 'PaymentChannelClaim':
-      tx.Flags = convertPaymentChannelClaimFlagsToNumber(tx.Flags)
+      tx.Flags = convertFlagsToNumber(tx.Flags, PaymentChannelClaimFlags)
       return
     case 'Payment':
-      tx.Flags = convertPaymentTransactionFlagsToNumber(tx.Flags)
+      tx.Flags = convertFlagsToNumber(tx.Flags, PaymentFlags)
       return
     case 'TrustSet':
-      tx.Flags = convertTrustSetFlagsToNumber(tx.Flags)
+      tx.Flags = convertFlagsToNumber(tx.Flags, TrustSetFlags)
       return
     default:
       tx.Flags = 0
   }
 }
 
-function convertAccountSetFlagsToNumber(
-  flags: AccountSetFlagsInterface,
-): number {
-  return reduceFlags(flags, AccountSetTfFlags)
-}
-
-function convertOfferCreateFlagsToNumber(
-  flags: OfferCreateFlagsInterface,
-): number {
-  return reduceFlags(flags, OfferCreateFlags)
-}
-
-function convertPaymentChannelClaimFlagsToNumber(
-  flags: PaymentChannelClaimFlagsInterface,
-): number {
-  return reduceFlags(flags, PaymentChannelClaimFlags)
-}
-
-function convertPaymentTransactionFlagsToNumber(
-  flags: PaymentFlagsInterface,
-): number {
-  return reduceFlags(flags, PaymentFlags)
-}
-
-function convertTrustSetFlagsToNumber(flags: TrustSetFlagsInterface): number {
-  return reduceFlags(flags, TrustSetFlags)
-}
-
 // eslint-disable-next-line @typescript-eslint/no-explicit-any -- added ValidationError check for flagEnum
-function reduceFlags(flags: GlobalFlags, flagEnum: any): number {
+function convertFlagsToNumber(flags: GlobalFlags, flagEnum: any): number {
```

Comment:
Should this change be in the listed changes since this seems like it was an existing function? (Or is it only used internally?)